daemons/fenced/fenced_remote.c

```diff
@@ -1035,17 +1011,17 @@ create_remote_stonith_op(const char *client, xmlNode * request, gboolean peer)
 
     if (op->call_options & st_opt_cs_nodeid) {
         int nodeid = crm_atoi(op->target, NULL);
-        char *nodename = stonith_get_peer_name(nodeid);
+        crm_node_t *node = crm_find_known_peer_full(nodeid, NULL, CRM_GET_PEER_ANY);
 
         /* Ensure the conversion only happens once */
         op->call_options &= ~st_opt_cs_nodeid;
 
-        if (nodename) {
+        if (node && node->uname) {
             free(op->target);
-            op->target = nodename;
+            op->target = strdup(node->uname);
 
         } else {
-            crm_warn("Could not expand nodeid '%s' into a host name", op->target);
+            crm_warn("Could not expand nodeid '%s' into a host name (%p)", op->target, node);
```

Comment:
Is the pointer value useful?

Reply:
Maybe not. It's from the reverting of c52267d ;-)